### DISCUSSION TITLE
A few questions about extracting pages from the database

### DISCUSSION THREAD

**Discussion by defnone** [lastorel/pytion#69] [Q&A] [answered]
1.     How can I retrieve a dictionary of all pages from the database with all page attributes?
2.     How to obtain a dictionary of pages where, for example, the status is 'in progress'?
3.     Is there a way to extract from the API only those database pages that have changed since the last update or are new?

Thank you.

**Comment by lastorel** [comment, marked as answer]
1. You can use `no.databases.db_query("DATABASE_ID")` method. It returns the PageArray with all pages, which includes all attributes.
```python
no = Notion(token=...)
pages = no.databases.db_query("DATABASE_ID")
# here is a list of pages
print(pages.obj)
print(pages.obj[0])
print(pages.obj[0].title)
print(pages.obj[0].properties["ATTRIBUTE_NAME"])
print(pages.obj[0].properties["ATTRIBUTE_NAME"].value)
```

2. Use 
```python
db = no.databases.get("DATABASE_ID")
db.db_filter(property_name="Status", property_type="status", value="in progress")
```
It will return the same type - Element with PageArray as in the example above

3. It was possible only for a specific database (not for all available databases) and relative to a specific time point with various conditions
```python
from datetime import datetime
db.db_filter(property_type="last_edited_time", condition="on_or_after", value=datetime(2023, 8, 27))
```
like that.
BUT, I found the issue - something changed on the backend of API. `last_edited_time` doesn't work at the moment

**Comment by lastorel** [comment]
#70
